elm-test/Main.elm

```diff
@@ -0,0 +1,90 @@
+module Main exposing (main)
+
+import Html exposing (..)
+import Html.Attributes exposing (..)
+import Html.Events exposing (onInput, onClick)
+
+type WeightUnit
+    = Pounds
+    | Kilos
+
+type alias WorkoutSet =
+    { reps : List Int
+    , weight : Int
+    , unitOfMeasurement : WeightUnit
+    }
+
+type Msg
+    = RepsUpdated String
+    | WeightUpdated String
+    | UnitUpdated WeightUnit
+
+initialWorkoutSet : WorkoutSet
+initialWorkoutSet =
+    { reps = []
+    , weight = 0
+    , unitOfMeasurement = Kilos
+    }
+
+view : WorkoutSet -> Html Msg
+view set =
+    div []
+        [ input [type_ "text", placeholder "reps", onInput RepsUpdated ] []
+        , input [type_ "text", placeholder "weight", onInput WeightUpdated] []
+        , select [ onInput (UnitUpdated << parseUnit) ]
```

Comment:
I think most browsers don't emit an "input" event when an option is selected. Instead, you probably want to listen to the "change" event. Unfortunately, `onChange` is not implemented by `Html.Events`.

It's a one-liner to write yourself though. I wrote an article about how to do so https://robots.thoughtbot.com/building-custom-dom-event-handlers-in-elm. Scroll all the way to the bottom if you just want to see the final implementation.

Reply:
Ah, shit. This does work in Safari and Chrome but doesn't work in Firefox. I had read that this was a new thing in 0.18, but I guess it's not 100% yet? I was trying to avoid needing to do the JSON decoding stuff. Oh well. Thanks for the link, I'll make this update.